hcipy/util/finite_difference.py

```diff
@@ -18,9 +20,9 @@ def generate_convolution_matrix(grid, kernel):
 	array_like
 		The matrix that applies the convolution.
 	'''
-	num_x = kernel.shape[1]
-	num_y = kernel.shape[0]
-
+	num_x = kernel.grid.shape[1]
+	num_y = kernel.grid.shape[0]
+	
 	yy, xx = np.meshgrid(np.arange(num_y), np.arange(num_y))
 	offsets = ((xx - num_x // 2) + (yy - num_y // 2) * grid.shape[0] ).ravel()
 	convolution_matrix = sparse.diags(kernel, offsets, shape=(grid.size, grid.size))
```

Comment:
Any check on whether `kernel.grid` has the same sample spacing (ie. `grid.delta`) as `grid`?

Reply:
Done.